buildSrc/src/main/groovy/io.deephaven.java-toolchain-conventions.gradle

```diff
@@ -123,30 +123,54 @@ def devJvmArgs = [
 //  '-XX:+PrintCompilation',          // this optional line shows jit operations as they happen
 ]
 
+def createVmOptions = tasks.register('createVmOptions') {
+  def vmOptionsFile = project.layout.buildDirectory.file('dh-default.vmoptions')
+  it.outputs.file(vmOptionsFile)
+  doFirst {
+    // These are supposed to be generally applicable and recommended JVM options, but they aren't hard requirements.
+    // Overly specific options do *not* belong here. For example, we should _not_ be setting something like `-Xmx4g` here.
+    // If you are tempted to try and put system properties here (`-Dkey=value`), think again; there should be a more
+    // appropriate place to set those.
+    //
+    // From the perspective of our application distribution, these options will be used as defaults for JAVA_OPTS
+    // (if the user already has JAVA_OPTS set, these VM options will _not_ apply).
+    vmOptionsFile.get().asFile.text = "-XX:+UseG1GC\n-XX:MaxGCPauseMillis=100\n-XX:+UseStringDeduplication\n"
+  }
+}
+
 tasks.withType(JavaExec).configureEach {
   def compilerDirectivesFile = createCompilerDirectives.get().outputs.files
+  def vmOptionsFile = createVmOptions.get().outputs.files
   inputs.files compilerDirectivesFile
+  inputs.files vmOptionsFile
   javaLauncher.set runtimeLauncher
-  jvmArgs += compilerArgs(compilerDirectivesFile.singleFile.path) + devJvmArgs
+  // Note: we _could_ have the vmOptionsFile constituents directly listed instead of using -XX:VMOptionsFile.
+  // That said, the current approach used here more closely matches how the application start script is defined.
+  jvmArgs += compilerArgs(compilerDirectivesFile.singleFile.path) + ["-XX:VMOptionsFile=${vmOptionsFile.singleFile.path}"] + devJvmArgs
 }
 
 tasks.withType(Test).configureEach {
   def compilerDirectivesFile = createCompilerDirectives.get().outputs.files
+  def vmOptionsFile = createVmOptions.get().outputs.files
   inputs.files compilerDirectivesFile
-
+  inputs.files vmOptionsFile
   javaLauncher.set testRuntimeLauncher
-  jvmArgs += compilerArgs(compilerDirectivesFile.singleFile.path) + devJvmArgs
+  // Note: we _could_ have the vmOptionsFile constituents directly listed instead of using -XX:VMOptionsFile.
+  // That said, the current approach used here more closely matches how the application start script is defined.
+  jvmArgs += compilerArgs(compilerDirectivesFile.singleFile.path) + ["-XX:VMOptionsFile=${vmOptionsFile.singleFile.path}"] + devJvmArgs
 }
 
 tasks.withType(GroovyCompile).configureEach {
   javaLauncher.set groovyCompilerLauncher
 }
 
 plugins.withType(ApplicationPlugin) {
-  applicationDistribution.into('lib') {
+  applicationDistribution.into('etc') {
```

Comment:
Why `etc` instead of `lib`?

Reply:
It's a better layout IMO, not mixing these files amongst the lib/*.jars. That said, I can easily revert this change.

```
$ find . -not -name "*.jar" 
.
./server-jetty
./server-jetty/lib
./server-jetty/bin
./server-jetty/bin/start
./server-jetty/bin/start.bat
./server-jetty/etc
./server-jetty/etc/dh-compiler-directives.txt
./server-jetty/etc/dh-default.vmoptions
```

I suspect in the future we may want to simplify how we declare our classpath. Right now it's a really long list, and it might be wise to do:

```
CLASSPATH=$APP_HOME/lib/*
```

in our start script. (Note this construction is not a shell glob - it's passed to the JVM w/ a `*` and the JVM works out the classpath.) It's a bit sloppy to have this match against the files shown here in etc.

I think we might find we also want to publish other "non-essential" bits in the future - maybe python requirements.txt files to say "these are the exact libraries we tested against". I think laying the groundwork for these potential bits makes sense.

Reply:
I'd probably vote against `*` on the classpath value, gradle's generated list gives us consistent ordering, and I don't think we're free of multiple jars containing the same-named classes yet.

The more I look at it, the more I like the etc/lib split, just in case something tries to load from classpath and gets configs - app.d stuff probably should be kept away from configs in the same way, to ensure that a groovy "class" there can't grab on to other files except by explicitly using the filesystem.